ctci/chapter2/2.6.py

```diff
@@ -0,0 +1,66 @@
+#2.6
+#Palindrome: Implement a function to check if a linked list is a palindrome.
+
+#assuming no cycles
+
+from random import randrange
+from collections import deque
+
+class Node:
+    def __init__(self, dataval=None):
+        self.dataval = dataval
+        self.nextval = None
+
+class SLinkedList:
+    def __init__(self):
+        self.headval = None
+    
+    @classmethod
+    def genList(cls, len :int, max: int):
+        l = SLinkedList()
+        l.headval = Node(randrange(max))
+        it = l.headval
+
+        for i in range(len):
+            n = Node (randrange(max))
+            it.nextval = n
+            it = it.nextval
+        return l
+
+    def printList(self):
+        s = ""
+        it = self.headval
+        while (it != None):
+            s += str((it.dataval)) + ", "
+            it = it.nextval
+        print(s)
+
+    def isPalindrome(self):
+        q = deque()
+        def iterator (n: Node):
+            nonlocal q
+            if (n == None):
+                return True
+            else:
+                q.append(n.dataval)
+                rtn = iterator(n.nextval)
```

Comment:
now, that I understand your nonlocal shenanigans, this type of programming makes more sense. But idk yucky